homework/weak_ptr/WeakPtr.hpp

```diff
@@ -26,50 +26,50 @@ class WeakPtr {
 
 private:
     T* rawPtr_;
-    ControlBlock<T>* ControlBlock_;
+    ControlBlock<T>* controlBlock_;
 };
 
 template <typename T>
 void WeakPtr<T>::controlBlockRemover() {
-    if (ControlBlock_->sharedRefsCounter_ == 0 && ControlBlock_->weakRefsCounter_ == 0) {
-        delete ControlBlock_;
+    if (controlBlock_->sharedRefsCounter_ == 0 && controlBlock_->weakRefsCounter_ == 0) {
+        delete controlBlock_;
     }
 }
 
 template <typename T>
 void WeakPtr<T>::weakCounterDecrementer() {
-    ControlBlock_->weakRefsCounter_.exchange(ControlBlock_->weakRefsCounter_.load(std::memory_order_relaxed) - 1,
+    controlBlock_->weakRefsCounter_.exchange(controlBlock_->weakRefsCounter_.load(std::memory_order_relaxed) - 1,
```

Comment:
```suggestion
    controlBlock_->weakRefsCounter_--
```
simple decrementation and incrementation works on atomics